maint/GenerateUcd.py

```diff
@@ -855,6 +862,17 @@ def write_bitsets(list, item_size):
   f.write('  NOTACHAR,\n')
 f.write('};\n\n')
 
+# --- Output the indices of the Turkish caseless character sets ---
+
+f.write("""\
+/* These are the indices, within ucd_caseless_sets, of the additional
+Turkish case-equivalences. */
+
+const uint32_t PRIV(ucd_turkish_dotted_i_caseset) = %d;
+const uint32_t PRIV(ucd_turkish_dotless_i_caseset) = %d;
```

Comment:
I don't feel very happy about adding these.

I wanted a simple and vaguely efficient way to transform `/[i]/i` → `/[iİ]/` at compile-time, so I wouldn't need to change anything in the matcher. How to neatly represent a set of exactly two characters? I thought I could re-use the existing PT_CLIST mechanism.

It's vaguely tidy. But even better would be an "OP_2CHAR" that gives a dedicated efficient encoding for either-of-two characters.

Reply:
Using PT_CLIST is a good idea.

Reply:
In another thread, I discussed getting rid of PT_CLIST, in which case I'd replace it with a simpler "OP_2CHAR".

Reply:
I would prefer a new `OP_VCHARI` that would point to the correct entry on this table depending of PCRE2_EXTRA_TURKISH_CASING. They are always 2 characters equivalences anyway while the casesets was meant to be used by 3 or more.

Reply:
The problem is, that each of these basic opcodes requires repeated variants, and that is already a lot of code cloning in the matcher. Implementing it is easy in jit, but we should not forget about the interpreter.

Reply:
It would be worth replacing PT_CLIST with "OP_2CHAR" in a future PR. It's not necessary right now though, just to get the Turkish feature.

Reply:
I hope a 2CHAR could also replace CHARI. Should be faster than reading UCD records. The CHARI could be limited to 8 bit only.